src/corehost/common/pal.unix.cpp

```diff
@@ -83,12 +86,22 @@ bool pal::is_path_rooted(const pal::string_t& path)
 bool pal::get_default_packages_directory(pal::string_t* recv)
 {
     recv->clear();
-    if (!pal::getenv("HOME", recv))
+    pal::string_t dir;
+    if (!pal::getenv("HOME", &dir))
+    {
+        struct passwd* pw = getpwuid(getuid());
+        if (pw && pw->pw_dir)
+        {
+            dir.assign(pw->pw_dir);
+        }
+    }
+    if (dir.empty())
     {
         return false;
     }
-    append_path(&*recv, _X(".nuget"));
-    append_path(&*recv, _X("packages"));
+    append_path(&dir, _X(".nuget"));
+    append_path(&dir, _X("packages"));
+    recv->assign(dir);
```

Comment:
Apparently NuGet is not resilient to HOME not being present. So I believe we need it if they do `dotnet restore`

Cc @yishaigalatzer